tests/stress_mt.c

```diff
@@ -138,21 +221,26 @@ static int test_multi_init(int enumerate)
 				tinfo[t].iteration,
 				libusb_error_name(tinfo[t].err));
 		} else if (enumerate) {
-			if (tinfo[t].devcount < 0) {
-				errs++;
-				fprintf(stderr,
-					"Thread %d failed to enumerate devices (iteration %d)\n",
+			if (t > 0 && tinfo[t].devcount != last_devcount) {
+				devcount_mismatch++;
+				printf("Device count mismatch: Thread %d discovered %ld devices instead of %ld\n",
 					tinfo[t].number,
-					tinfo[t].iteration);
-			} else {
-				printf("Thread %d discovered %ld devices\n",
-				       tinfo[t].number,
-				       (long int) tinfo[t].devcount);
+					(long int) tinfo[t].devcount,
```

Comment:
Why are those casts necessary btw? Wouldn't it be easier to change the format string?

Reply:
IIRC some compilers (on non-Posix platforms) don't have the %z support for ssize_t.

Reply:
On Debian (bookworm) with gcc version 12 this is no longer a problem. I use the latter for compile testing, but the former is my daily driver so I'd like to keep it supported.

Reply:
Ah I thought device count fields are just an int, didn't notice ssize_t. Going over int limits is unrealistic anyway.